cs/src/Management/TunnelManagementClient.cs

```diff
@@ -804,53 +757,14 @@ public void Dispose()
             Tunnel tunnel,
             string? path,
             string? query,
-            TunnelRequestOptions? options)
-        {
-            Requires.NotNull(tunnel, nameof(tunnel));
-
-            string tunnelPath;
-            var pathBase = TunnelsPath;
-            if (!string.IsNullOrEmpty(tunnel.ClusterId) && !string.IsNullOrEmpty(tunnel.TunnelId))
-            {
-                tunnelPath = $"{pathBase}/{tunnel.TunnelId}";
-            }
-            else
-            {
-                Requires.Argument(
-                    !string.IsNullOrEmpty(tunnel.Name),
-                    nameof(tunnel),
-                    "Tunnel object must include either a name or tunnel ID and cluster ID.");
-
-                if (string.IsNullOrEmpty(tunnel.Domain))
-                {
-
-                    tunnelPath = $"{pathBase}/{tunnel.Name}";
-                }
-                else
-                {
-                    // Append the domain to the tunnel name.
-                    tunnelPath = $"{pathBase}/{tunnel.Name}.{tunnel.Domain}";
-                }
-            }
-
-            return BuildUri(
-                tunnel.ClusterId,
-                tunnelPath + (!string.IsNullOrEmpty(path) ? path : string.Empty),
-                query,
-                options);
-        }
-
-        private Uri BuildTunnelUri(
-            TunnelV2 tunnel,
-            string? path,
-            string? query,
-            TunnelRequestOptions? options)
+            TunnelRequestOptions? options,
+            bool isCreate = false)
         {
             Requires.NotNull(tunnel, nameof(tunnel));
 
             string tunnelPath;
             var pathBase = TunnelsPath;
-            if (!string.IsNullOrEmpty(tunnel.ClusterId) && !string.IsNullOrEmpty(tunnel.TunnelId))
+            if (!string.IsNullOrEmpty(tunnel.ClusterId) && !string.IsNullOrEmpty(tunnel.TunnelId) || isCreate)
             {
                 tunnelPath = $"{pathBase}/{tunnel.TunnelId}";
```

Comment:
If `isCreate` is true and `TunnelId` is null, the `tunnelPath` will be incorrect.

Reply:
added additional check